API/RestAPI.tsx

```diff
@@ -0,0 +1,20 @@
+import { WPComAPIVersion } from "./APIs";
+
+/*
+ * Utility function to create a rest request using hitting the site url directly.
+ */
+export async function restFetch(
+  apiVersion: WPComAPIVersion,
+  path: string,
+  siteUrl: string,
+  appPassword: string
+) {
+  const url = `${siteUrl}/wp-json/${apiVersion}/${path}`;
+  console.log(`-- About to fetch: ${url}`); // We can delete this but seems handy for the time being to seee what requests are being fired.
+
+  return fetch(url, {
+    headers: {
+      Authorization: `Basic ${appPassword}`,
```

Comment:
I try to understand how this works - when I provide site and appPassword, I get 401 error.

On Android, when we use Application Password authentication, we also add a username and then we encode it to base64. Don't we have to do the same here?

<img width="536" alt="image" src="https://github.com/woocommerce/WooCommerce-Shared/assets/5845095/cf0a3cf8-3ce1-4a07-a07a-759866cbf53b">

Reply:
I was only able to authenticate using Application Passwords when added changes:
- add `<username>:` prefix to the password
- encode via base64

so this matches Android. I've prepared a PR to discuss those changes here: https://github.com/woocommerce/WooCommerce-Shared/pull/53

Reply:
Considering that "application password" in the context of this PR is encoded by base64 "username:password" format, everything works well: 

https://github.com/woocommerce/WooCommerce-Shared/assets/5845095/0c248fdc-3db5-4857-9424-35f4ac0d0da2

I was confused a little because I didn't work with application password before so I wasn't sure what's what. So I'll close #53 and merge this PR as things work well 👍